src/ap.js

```diff
@@ -19,8 +19,14 @@ var map = require('./map');
  *
  *      R.ap([R.multiply(2), R.add(3)], [1,2,3]); //=> [2, 4, 6, 4, 5, 6]
  */
-module.exports = _curry2(function ap(fns, vs) {
-  return _hasMethod('ap', fns) ? fns.ap(vs) : _reduce(function(acc, fn) {
-    return _concat(acc, map(fn, vs));
-  }, [], fns);
+module.exports = _curry2(function ap(functor, fn) {
+  return _hasMethod('ap', functor) ?
+           functor.ap(fn) :
+         functor instanceof Function ?
+           function(x) {
+             return functor(x)(fn(x));
+           } :
+           _reduce(function(acc, f) {
+             return _concat(acc, map(f, fn));
+           }, [], functor);
 });
```

Comment:
Perhaps this would read better formatted like so:

```javascript
module.exports = _curry2(function ap(ftor, fn) {
  return (
    _hasMethod('ap', ftor) ?
      ftor.ap(fn) :
    ftor instanceof Function ?
      function(x) { return ftor(x)(fn(x)); } :
    // else
      _reduce(function(acc, f) { return _concat(acc, map(f, fn)); }, [], ftor)
  );
});
```

Also, let's use `typeof ftor === 'function'` rather than `ftor instanceof Function` for the reason given in #1140.